code_submitter/auth.py

```diff
@@ -199,3 +201,45 @@ def __init__(self, data: List[NemesisUserInfo] = DEFAULT) -> None:
 
     async def load_user(self, username: str, password: str) -> NemesisUserInfo:
         return self.data[username]
+
+
+class FileBackend(BasicAuthBackend):
+    """
+    Authentication backend which stores credentials in a YAML file.
+
+    Credentials are stored in the format `TLA: password`.
```

Comment:
Probably worth explicitly noting that we're storing the passwords in plaintext and why that's an acceptable thing to do for this use-case.